accelerator/core/core.c

```diff
@@ -780,20 +795,37 @@ status_t ta_fetch_txn_with_uuid(const ta_cache_t* const cache, const char* const
   if (exist) {
     res->status = SENT;
 
-    ret = cache_get(uuid, (char*)txn_flex_trits);
+    int len = 0;
+    ret = cache_list_size(uuid, &len);
     if (ret) {
       ta_log_error("%s\n", ta_error_to_string(ret));
       goto done;
     }
 
-    res->txn = transaction_deserialize(txn_flex_trits, false);
-    if (res->txn == NULL) {
-      ret = SC_CORE_OOM;
+    for (int i = 0; i < len; ++i) {
+      flex_trit_t txn_flex_trits[NUM_FLEX_TRITS_SERIALIZED_TRANSACTION + 1];
+      ret = cache_list_at(uuid, i, NUM_FLEX_TRITS_SERIALIZED_TRANSACTION, (char*)txn_flex_trits);
+      if (ret) {
+        ta_log_error("%s\n", ta_error_to_string(ret));
+        goto done;
+      }
+      iota_transaction_t* txn = transaction_deserialize(txn_flex_trits, false);
+
+      if (bundle_transactions_add(res->bundle, txn) != RC_OK) {
+        ret = SC_CORE_NULL;
+        ta_log_error("%s\n", "Failed to add transaction to bundle.");
+        goto done;
+      }
+      free(txn);
+    }
```

Comment:
Is this possible that `res->bundle` equals to `NULL` and `txn` does not?
If it is, then you forget to free the `txn`.

Reply:
res->bundle may not be NULL, but txn could be. I will free txn in the if.